proc/combine/combine.go

```diff
@@ -0,0 +1,90 @@
+package combine
+
+import (
+	"sync"
+
+	"github.com/brimsec/zq/proc"
+	"github.com/brimsec/zq/zbuf"
+)
+
+// A Merge proc merges multiple upstream inputs into one output.
+type Proc struct {
+	pctx     *proc.Context
+	once     sync.Once
+	ch       <-chan proc.Result
+	doneCh   chan struct{}
+	parents  []*runnerProc
+	nparents int
+}
+
+type runnerProc struct {
+	pctx   *proc.Context
+	parent proc.Interface
+	ch     chan<- proc.Result
+	doneCh <-chan struct{}
```

Comment:
Nit: You could simplify a tiny bit if you replaced pctx and doneCh with a context.Context created by context.WithCancel.

Reply:
This was the same change Nibs suggested in the previous orderedmerge code I moved to zbuf, so yes, good idea, I will change it here too (as this was an unmodified copy except for name from the old merge proc).